### PR TITLE
Implement CLZ opcode

### PR DESCRIPTION
**Pull request by elopez**
## Description
This also bumps the Solidity version used in tests to be able to compile an example with CLZ

## Checklist

- [x] tested locally
- [x] added automated tests
- [ ] updated the docs
- [x] updated the changelog

### CODE REVIEW

**Comment by elopez** [comment]
It is worth noting that this does not implement CLZ on the SMT, help is welcome on that regard 😄

File: src/EVM/Expr.hs [hunk 1]
```diff
@@ -1236,6 +1239,9 @@ simplifyNoLitToKeccak e = untilFixpoint (mapExpr go) e
        | a == (Lit 0) = v
        | v == (Lit 0) = v
        | otherwise = sar a v
+    go (CLZ v)
+       | v == (Lit 0) = Lit 256
```
Comment (msooseth) [inline src/EVM/Expr.hs]: There is no reason for this -- `clz` will do that itself, so I'm removing.

**Comment by msooseth** [comment]
OK, this should now work, but the ethereum-tests have moved, and have changed format, so that needs to be updated, as per @elopez

**Comment by msooseth** [comment]
@elopez wanna review this? I resolved the cash with main by taking the `main` on the flake.lock, I hope that's good? The change is: https://github.com/argotorg/hevm/pull/940/commits/84c1e37bd308b8ce3cfc0efb6bbf1900e9def732 Plus I added a CLZ smt implementation of course

File: test/contracts/pass/fusaka.sol
Comment (blishko) [inline test/contracts/pass/fusaka.sol:1]: Maybe we should add a negative test as well?

File: src/EVM/SMT/SMTLIB.hs [hunk 2]
```diff
@@ -203,8 +203,44 @@ prelude =  SMT2 src mempty mempty
...
       \(ite (= b (_ bv30 256)) ((_ sign_extend 8  ) ((_ extract 247  0) val)) val)\
-      \)))))))))))))))))))))))))))))))"
-    ]
+      \)))))))))))))))))))))))))))))))",
+    "(define-fun clz256 ((x (_ BitVec 256))) (_ BitVec 256)\
+      \(let ((x128 (ite (bvuge x   #x0000000000000000000000000000000100000000000000000000000000000000)\
+      \                 (bvlshr x    (_ bv128 256)) x))\
+      \      (n128 (ite (bvuge x   #x0000000000000000000000000000000100000000000000000000000000000000)\
+      \               (_ bv0 256) (_ bv128 256))))\
+      \(let ((x64 (ite (bvuge  x128 #x0000000000000000000000000000000000000000000000010000000000000000)\
```
Comment (blishko) [inline src/EVM/SMT/SMTLIB.hs]:
```suggestion
      \(let ((x64 (ite (bvuge x128 #x0000000000000000000000000000000000000000000000010000000000000000)\
```
Just to have it aligned nicely.

**Comment by elopez** [comment]
> @elopez wanna review this? I resolved the cash with main by taking the `main` on the flake.lock, I hope that's good? The change is: [84c1e37](https://github.com/argotorg/hevm/commit/84c1e37bd308b8ce3cfc0efb6bbf1900e9def732) Plus I added a CLZ smt implementation of course

IIRC the flake.lock change was upgrading the solcs.nix, but I think that might have been upgraded independently since opening this PR. As long as we have a recent enough solc that can do `clz` we should be ok :)

**Comment by elopez** [comment]
@msooseth can you remove tests that should pass now from the problematicTests list? It should be at least these, but possibly some other ones as well 😄 https://github.com/argotorg/hevm/blob/17e0724/test/EVM/Test/BlockchainTests.hs#L217-L221

File: src/EVM/SMT/SMTLIB.hs (hunk 2 above)
Comment (msooseth) [reply src/EVM/SMT/SMTLIB.hs]: I have now fixed the alignment everywhere. Should be better now.

File: test/contracts/pass/fusaka.sol
Comment (msooseth) [reply test/contracts/pass/fusaka.sol:1]: OK, added more tests, including negative tests. I actually much prefer these to the ones that were added as part of forge, because forge tests are slower and harder to edit. But I'll leave those in, too, they do check forge which is nice.

**Comment by msooseth** [comment]
Damn. Windows is failing, maybe because of the nix "cleanup" I did:
```
2026-02-04T14:12:08.1858158Z     Fusaka:                                                       compilation failed: exit code: ExitFailure 1
2026-02-04T14:12:08.1859277Z 
2026-02-04T14:12:08.1859664Z stdout:
2026-02-04T14:12:08.1859843Z 
2026-02-04T14:12:08.1859850Z 
2026-02-04T14:12:08.1859968Z stderr:
2026-02-04T14:12:08.1860896Z Error: Encountered invalid solc version in src\unit-tests.t.sol: No solc version exists that matches the version requirement: ^0.8.31
2026-02-04T14:12:08.1861958Z 
2026-02-04T14:12:08.1945072Z FAIL
2026-02-04T14:12:08.1952906Z       Exception: Internal Error: Error compiling test file "test/contracts/pass/fusaka.sol" in directory "D:\\a\\_temp\\msys64\\tmp\\dapp-test-dc2ed3372c7debb4" -- CallStack (from HasCallStack):
```

**Comment by msooseth** [comment]
OK, moved Windows to latest solidity.
